tests/Unit/TestHelpers.hpp

```diff
@@ -138,6 +138,28 @@ void test_move_semantics(T&& a, const T& comparison) {
 #pragma GCC diagnostic pop
 }
 
+// Test for iterators
+template <typename Container>
+void test_iterators(const Container& c) {
+  CHECK(c.end() - c.begin() == c.size());
+  CHECK(c.begin() == c.cbegin());
+  CHECK(c.end() == c.cend());
+  CHECK(c.rend() - c.rbegin() == c.size());
+
+  auto it = c.begin();
```

Comment:
what about the other begin and end cases? This only checks half. We also should have a `test_iterators` and a `test_reverse_iterators` because unordered associative containers do not have reverse iteration since there is no order defined.